python/pyspark/sql/udf.py

```diff
@@ -439,13 +439,23 @@ def func(*args: Any, **kwargs: Any) -> Any:
                 func.__signature__ = inspect.signature(f)  # type: ignore[attr-defined]
                 judf = self._create_judf(func)
                 jPythonUDF = judf.apply(_to_seq(sc, cols, _to_java_column))
-                id = jPythonUDF.expr().resultId().id()
+                id = self._get_UDF_id(jPythonUDF.expr())
                 sc.profiler_collector.add_profiler(id, memory_profiler)
         else:
             judf = self._judf
             jPythonUDF = judf.apply(_to_seq(sc, cols, _to_java_column))
         return Column(jPythonUDF)
 
+    def _get_UDF_id(self, jexpr: JavaObject) -> int:
```

Comment:
I think should be lowercased

```suggestion
    def _get_udf_id(self, jexpr: JavaObject) -> int:
```